tests/core.py

```diff
@@ -116,13 +117,9 @@ def setUp(self):
 
     def tearDown(self):
         if os.environ.get('KUBERNETES_VERSION') is None:
-            session = Session()
-            session.query(models.TaskInstance).filter_by(
-                dag_id=TEST_DAG_ID).delete()
-            session.query(TaskFail).filter_by(
-                dag_id=TEST_DAG_ID).delete()
-            session.commit()
-            session.close()
+            with create_session() as session:
```

Comment:
How about `provide_session` annotation instead of `create_session` ?

Reply:
That would work here as well. I prefer the `create_session` in the `with` statement since it is more transparent to me when the session is being committed and closed.